ChangeSet
  1.2028 05/10/25 15:23:49 ingo@stripped +1 -0
  Bug#13707 - Server crash with INSERT DELAYED on MyISAM table
  Initialized 'ptr' for a newly instantiated varstring field.
  This is required by INSERT DELAYED.

  sql/field.cc
    1.286 05/10/25 15:23:41 ingo@stripped +10 -3
    Bug#13707 - Server crash with INSERT DELAYED on MyISAM table
    Initialized 'ptr' for a newly instantiated varstring field.
    This is required by INSERT DELAYED.

--- 1.285/sql/field.cc  2005-10-13 23:04:47 +02:00
+++ 1.286/sql/field.cc  2005-10-25 15:23:41 +02:00
@@ -6201,9 +6201,16 @@
     This is done to ensure that ALTER TABLE will convert old VARCHAR fields
     to now VARCHAR fields.
   */
-  return new Field_varstring(field_length, maybe_null(),
-                             field_name, new_table,
-                             charset());
+  Field *new_field= new Field_varstring(field_length, maybe_null(),
+                                        field_name, new_table,
+                                        charset());
+  /*
+    delayed_insert::get_local_table() needs a ptr copied from old table.
+    This is what other new_field() methods do too. The above method of
+    Field_varstring sets ptr to NULL.
+  */
+  new_field->ptr= ptr;
+  return new_field;
 }
 
 /****************************************************************************
